### PR TITLE
feat(example): add a music player example

### PR DESCRIPTION
**Pull request by gdorsi**
# Description

I'm creating a new example to show how the `BinaryCoStream` can be used to manage files and how the share API can be used to enable collaboration across multiple accounts.

### CODE REVIEW

File: examples/musicPlayer/src/1_schema.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    co,
+    Profile,
+    Account,
+  } from "jazz-tools";
+  
+  export class MusicTrack extends CoMap {
+    file = co.ref(BinaryCoStream);
+    title = co.string;
+    duration = co.number;
+    waveform = co.json<number[]>();
```
Comment (gdorsi) [inline examples/musicPlayer/src/1_schema.ts]: waveform is a list of numbers that won't never change.

Not sure if `co.json` is the better primitive for this kind of data.

File: examples/musicPlayer/src/3_useMediaPlayer.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
+import { usePlayMedia } from "@/lib/audio/usePlayMedia";
+import { usePlayState } from "@/lib/audio/usePlayState";
+import { useAccount, useCoState } from "@/lib/jazz";
+import { MusicTrack, Playlist } from "@/1_schema";
+import { BinaryCoStream } from "jazz-tools";
+import { useEffect } from "react";
+
+export function useMediaPlayer() {
+    const { me } = useAccount();
```
Comment (gdorsi) [inline examples/musicPlayer/src/3_useMediaPlayer.tsx:10]: I have some issues with this hook when using the `depth` param.

By calling the hook like this when trying a cold start the account never gets loaded:
```
const { me } = useAccount({
       root: {
            activePlaylist: {
                tracks: [{}],
            },
            activeTrack: {
                file: [],
            },
        },
});
```

Also it was causing some type errors on the below assignements:

```ts
me.root.activeTrack = next;
```

@aeplay is this a wrong usage of the API or a bug to report?

File: examples/musicPlayer/src/3_useMediaPlayer.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    const activeTrack = useCoState(MusicTrack, me?.root?.activeTrack?.id, {
+        file: [],
```
Comment (gdorsi) [inline examples/musicPlayer/src/3_useMediaPlayer.tsx]: This depth definition was required to have a non-null value on the `file` attribute.
Also passing an array here doesn't look right, but TS forced me into it.

Is this intentional?

File: examples/musicPlayer/src/1_schema.ts (hunk 1 above)
Comment (aeplay) [reply examples/musicPlayer/src/1_schema.ts]: if that's the case, that's indeed the best way to do it - just how huge is it? Is it worth putting it in a referenced sub-coMap so we can choose to only load it when needed?

Comment (gdorsi) [reply examples/musicPlayer/src/1_schema.ts]: > Is it worth putting it in a referenced sub-coMap so we can choose to only load it when needed?

Awesome idea!
Would be great way to show off some load optimizations 🚀

**Pull request closed without merging**